mcp-server/build.gradle.kts

```diff
@@ -0,0 +1,274 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.gradle.api.GradleException
+import org.gradle.api.tasks.Delete
+import org.gradle.api.tasks.Exec
+import org.gradle.kotlin.dsl.register
+import java.io.File
+
+val pythonProjectDir = project.projectDir
+val venvDir = pythonProjectDir.resolve(".venv")
+
+val globalUvExecutable = when {
+  System.getProperty("os.name").contains("win", ignoreCase = true) -> "uv.exe"
+  else -> "uv"
+}
+
+fun getUvExecutable(): String {
+  return if (project.extensions.extraProperties.has("uvExecutable")) {
+    project.extensions.extraProperties.get("uvExecutable") as String
+  } else {
+    globalUvExecutable
+  }
+}
+
+val venvPython = when {
+  System.getProperty("os.name").contains("win", ignoreCase = true) ->
+    venvDir.resolve("Scripts/python.exe").absolutePath
+  else ->
+    venvDir.resolve("bin/python").absolutePath
+}
+
+tasks {
+  register<Exec>("installUv") {
+    group = "python"
+    description = "Install UV if not present"
+
+    val isWindows = System.getProperty("os.name").contains("win", ignoreCase = true)
+    val uvInstallDir = file("$pythonProjectDir/.uv/bin").apply { mkdirs() }
+    val uvExecutable = if (isWindows) {
+      file("$uvInstallDir/uv.exe")
+    } else {
+      file("$uvInstallDir/uv")
+    }
+
+    onlyIf {
+      if (uvExecutable.exists()) {
+        project.extensions.extraProperties.set("uvExecutable", uvExecutable.absolutePath)
+        logger.lifecycle("UV already installed at: ${uvExecutable.absolutePath}")
+        false
+      } else {
+        try {
+          exec {
+            commandLine(globalUvExecutable, "--version")
+            isIgnoreExitValue = true
+          }.exitValue != 0
+        } catch (e: Exception) {
+          true
+        }
+      }
+    }
+
+    doFirst {
+      logger.lifecycle("UV not found, installing via official script...")
+    }
+
+    if (isWindows) {
+      commandLine(
+        "powershell",
+        "-Command",
+        "\$env:UV_INSTALL_DIR='${uvInstallDir.absolutePath}'; " +
+          "irm https://astral.sh/uv/install.ps1 | iex"
+      )
+    } else {
+      commandLine(
+        "/bin/sh",
+        "-c",
+        "export UV_INSTALL_DIR=${uvInstallDir.absolutePath}; " +
+          "curl -LsSf https://astral.sh/uv/install.sh | sh"
+      )
+    }
+
+    doLast {
+      val uvCheck = exec {
+        commandLine(uvExecutable.absolutePath, "--version")
+        isIgnoreExitValue = true
+      }
+      if (uvCheck.exitValue != 0) {
+        throw GradleException("UV installation failed")
+      }
+
+      project.extensions.extraProperties.set("uvExecutable", uvExecutable.absolutePath)
+    }
+  }
```

Comment:
What if I already installed a UV, will this install another one for me? Also, how about let user to install UV beforehand. My concern is that a): Gradle may not have the permission to install; b) User may want a different version. 

I don't have a strong preference on it, just some thougts about it.